### PR TITLE
feat/seed-enhance

### PR DESCRIPTION
**Pull request by kevinslin**
bag of improvements for seed bank:

- better cli output
- support custom `site` object
- copy note url is seed specific
- convert remove vaults into seeds

### CODE REVIEW

File: packages/common-all/src/types/seed.ts [hunk 1]
```diff
@@ -6,6 +6,15 @@ export type SeedConfig = {
...
   contact?: SeedContact;
+  /**
+   * Url for seed
+   */
+  site?: SeedSite;
+};
+
+export type SeedSite = {
+  url: string;
+  index?: string;
```
Comment (jonathanyeung) [inline packages/common-all/src/types/seed.ts:17]: What does index here mean/do?

File: packages/engine-server/src/seed/service.ts [hunk 2]
```diff
@@ -99,8 +104,24 @@ export class SeedService {
...
+    const spath = SeedUtils.seed2Path({ wsRoot, id });
+    fs.ensureDirSync(path.dirname(spath));
+    const git = simpleGit({ baseDir: wsRoot });
+    const options: {
+      "--track"?: string;
+    } = {};
+    if (branch) {
+      options["--track"] = branch;
+    }
+    await git.clone(seed.repository.url, spath, options);
```
Comment (jonathanyeung) [inline packages/engine-server/src/seed/service.ts:124]: Any possibilities of name collisions here? i.e. if folder w/ name already exists.

File: packages/dendron-cli/src/commands/base.ts [hunk 3]
```diff
@@ -47,11 +49,13 @@ export abstract class CLICommand<TOpts, TOut> extends BaseCommand<TOpts, TOut> {
...
-    // add wsRoot if not exist
     if (!args.wsRoot) {
       const cwd = process.cwd();
-      if (!fs.existsSync(path.join(cwd, "dendron.yml"))) {
-        console.log("no workspace deted. --wsRoot must be set");
+      if (
+        !fs.existsSync(path.join(cwd, "dendron.yml")) &&
+        !this.wsRootOptional
+      ) {
+        console.log("no workspace detecd. --wsRoot must be set");
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply packages/dendron-cli/src/commands/base.ts:58]: you are finding all my typos :)

File: packages/common-all/src/types/seed.ts (hunk 1 above)
Comment (kevinslin) [reply packages/common-all/src/types/seed.ts:17]: equivalent of `siteIndex`, see https://wiki.dendron.so/notes/f2ed8639-a604-4a9d-b76c-41e205fb8713.html#siteindex-optional

File: packages/engine-server/src/seed/service.ts (hunk 2 above)
Comment (kevinslin) [reply packages/engine-server/src/seed/service.ts:124]: we are going with the npm approach where the name has to be globally unique. a seed name is `{publisher}.{name}`

**Review by jonathanyeung — APPROVED** [review]

**Pull request closed without merging**